### DISCUSSION TITLE
Weird bus error

### DISCUSSION THREAD

**Discussion by al7aro** [raysan5/raylib#2984] [Q&A]
Hey comunity!!!!
So I have been trying to compile my raylib project using cmake because I work together on macOS M1 and windows, and it is kind of working.
BUT after hours of getting a `bus error` on macOS I came into this situation:
I create this simple project:
```
.
├── CMakeLists.txt
├── build/
├── raylib/
├── main.c
└── raylib.h
```
Where the `raylib` folder is simply a clone of the `raylib` repo and `CMakeLists.txt` is:
```
cmake_minimum_required(VERSION 3.24)
project(Rayplot)
add_executable(rayplot main.c)
add_subdirectory(raylib)
target_link_libraries(rayplot raylib)
```
Now here comes the problem.
Using the following `main.c` works just fine:
```
#include "./raylib.h"

int	main()
{
	InitWindow(500, 500, "TITLE");
	while (!WindowShouldClose())	
	{
		BeginDrawing();
		ClearBackground(WHITE);
		EndDrawing();
	}
	CloseWindow();
	return (0);
}
```
But, as soon as I simply create a GLOBAL VARIABLE called `time` (it must be called `time` lower case for the error to occur) this happens:
```
#include "./raylib.h"
double time;
int	main()
{
	InitWindow(500, 500, "TITLE");
	while (!WindowShouldClose())	
	{
		BeginDrawing();
		ClearBackground(WHITE);
		EndDrawing();
	}
	CloseWindow();
	return (0);
}
```
The output is:
`[1]    31848 bus error  ./rayplot`
The rest of the output is the normal output raylib gives.

I have tested it on Windows too and there isn't any problem.
It only happens when compiling from sources, not with the `brew` version.
I have tried with some variable names and types but it only happens with `time` and I haven't found anything online.
Do you know why this happens? Is this something about raylib or glfw? Is there a way to solve it or is it simply a library bug?

Thank you anyway. <3

**Comment by chriscamacho** [comment]
try compiling with -g and running it through gdb, a proper backtrace will likely give you some insight

**Comment by Peter0x44** [comment]
The reason this happens is because `time` is a Linux syscall, so the name is therefore reserved. I don't know exactly the mechanism why this crash happens exactly, I personally got an ordinary segfault, no "bus error". I'm guessing a function pointer of sorts gets overwritten by your global variable.

It gets called by InitTimer in InitWindow, which then calls glfwGetTime, and then crashes somewhere within, when that syscall gets invoked.

The same thing happens with other names - I've seen one other person have a similar strange result with `select`, which is also a Linux syscall that then gets used by some code in glfw later. Consider all Linux syscall names reserved, and don't name your global variables or functions the same.

The solution is simple - name your variables better.
For global variables, a common convention is to prefix them with g_.

I'm pretty sure there is nothing raylib can do to avoid this, so I wouldn't call it a raylib bug. I could be mistaken about this, though.

See `man syscalls` for all the other names of Linux syscalls you shouldn't conflict with.

**Comment by orcmid** [comment]
I think the conflict arises because at the top level, the `double time;` declaration is automatically `extern`.  This leads to potential conflicts with standard or platform libraries.  

To avoid conflicts such as that, as well as colliding with system externals that may be linked in/by some silently-used libraries, use either `static double time;` or, best, simply move the declaration inside the `main()` code.
